implementations/micrometer-registry-prometheus/src/main/java/io/micrometer/prometheus/PrometheusNamingConvention.java

```diff
@@ -53,20 +53,12 @@ public PrometheusNamingConvention(String timerSuffix) {
     public String name(String name, Meter.Type type, @Nullable String baseUnit) {
         String conventionName = NamingConvention.snakeCase.name(name, type, baseUnit);
 
-        switch (type) {
-            case COUNTER:
-            case DISTRIBUTION_SUMMARY:
-            case GAUGE:
-                if (baseUnit != null && !conventionName.endsWith("_" + baseUnit))
-                    conventionName += "_" + baseUnit;
-                break;
-        }
-
         switch (type) {
             case COUNTER:
```

Comment:
That's not the same for counter. For counter it would do 

```java
                    conventionName += "_" + baseUnit;
```

and then

```
if (!conventionName.endsWith("_total"))
                    conventionName += "_total";
```

whereas in your case it does only the conventionName part.